include/zephyr/drivers/mfd/max22017.h

```diff
@@ -0,0 +1,141 @@
+#define MAX22017_GEN_INT_TMOUT_INT         BIT(8)
+#define MAX22017_GEN_INT_THSHDN_INT        GENMASK(7, 6)
+#define MAX22017_GEN_INT_THWRNG_INT        GENMASK(5, 4)
+#define MAX22017_GEN_INT_OVC_INT           GENMASK(3, 2)
+#define MAX22017_GEN_INT_CRC_INT           BIT(1)
+#define MAX22017_GEN_INT_GPI_INT           BIT(0)
+
+#define MAX22017_GEN_INTEN_OFF                 0x09
+#define MAX22017_GEN_INTEN_CONV_OVF_INTEN      GENMASK(13, 12)
+#define MAX22017_GEN_INTEN_OPENWIRE_DTCT_INTEN GENMASK(11, 10)
+#define MAX22017_GEN_INTEN_HVDD_INTEN          BIT(9)
+#define MAX22017_GEN_INTEN_TMOUT_INTEN         BIT(8)
+#define MAX22017_GEN_INTEN_THSHDN_INTEN        GENMASK(7, 6)
+#define MAX22017_GEN_INTEN_THWRNG_INTEN        GENMASK(5, 4)
+#define MAX22017_GEN_INTEN_OVC_INTEN           GENMASK(3, 2)
+#define MAX22017_GEN_INTEN_CRC_INTEN           BIT(1)
+#define MAX22017_GEN_INTEN_GPI_INTEN           BIT(0)
+
+#define MAX22017_GEN_RST_CTRL_OFF             0x0A
+#define MAX22017_GEN_RST_CTRL_AO_COEFF_RELOAD GENMASK(15, 14)
+#define MAX22017_GEN_RST_CTRL_GEN_RST         BIT(9)
+
+#define MAX22017_AO_CMD_OFF        0x40
+#define MAX22017_AO_CMD_AO_LD_CTRL GENMASK(15, 14)
+
+#define MAX22017_AO_STA_OFF      0x41
+#define MAX22017_AO_STA_BUSY_STA GENMASK(15, 14)
+#define MAX22017_AO_STA_SLEW_STA GENMASK(13, 12)
+#define MAX22017_AO_STA_FAIL_STA BIT(0)
+
+#define MAX22017_AO_CNFG_OFF                  0x42
+#define MAX22017_AO_CNFG_AO_LD_CNFG           GENMASK(15, 14)
+#define MAX22017_AO_CNFG_AO_CM_SENSE          GENMASK(13, 12)
+#define MAX22017_AO_CNFG_AO_UNI               GENMASK(11, 10)
+#define MAX22017_AO_CNFG_AO_MODE              GENMASK(9, 8)
+#define MAX22017_AO_CNFG_AO_OPENWIRE_DTCT_LMT GENMASK(5, 4)
+#define MAX22017_AO_CNFG_AI_EN                GENMASK(3, 2)
+#define MAX22017_AO_CNFG_AO_EN                GENMASK(1, 0)
+
+#define MAX22017_AO_SLEW_RATE_CHn_OFF(n)                (0x43 + n)
+#define MAX22017_AO_SLEW_RATE_CHn_AO_SR_EN_CHn          BIT(5)
+#define MAX22017_AO_SLEW_RATE_CHn_AO_SR_SEL_CHn         BIT(4)
+#define MAX22017_AO_SLEW_RATE_CHn_AO_SR_STEP_SIZE_CHn   GENMASK(3, 2)
+#define MAX22017_AO_SLEW_RATE_CHn_AO_SR_UPDATE_RATE_CHn GENMASK(1, 0)
+
+#define MAX22017_AO_DATA_CHn_OFF(n)     (0x45 + n)
+#define MAX22017_AO_DATA_CHn_AO_DATA_CH GENMASK(15, 0)
+
+#define MAX22017_AO_OFFSET_CORR_CHn_OFF(n)       (0x47 + (2 * n))
+#define MAX22017_AO_OFFSET_CORR_CHn_AO_OFFSET_CH GENMASK(15, 0)
+
+#define MAX22017_AO_GAIN_CORR_CHn_OFF(n)     (0x48 + (2 * n))
+#define MAX22017_AO_GAIN_CORR_CHn_AO_GAIN_CH GENMASK(15, 0)
+
+#define MAX22017_SPI_TRANS_ADDR    GENMASK(7, 1)
+#define MAX22017_SPI_TRANS_DIR     BIT(0)
+#define MAX22017_SPI_TRANS_PAYLOAD GENMASK(15, 0)
+
+/**
+ * @defgroup mdf_interface_max22017 MFD MAX22017 interface
+ * @ingroup mfd_interfaces
+ * @{
+ */
+
+int max22017_reg_read(const struct device *dev, uint8_t addr, uint16_t *value);
+int max22017_reg_write(const struct device *dev, uint8_t addr, uint16_t value);
```

Comment:
I know they are self-explanatory however you could still consider documenting these two functions.